### DISCUSSION TITLE
サーボモータのM5Stack接続側のケーブルは各自で加工するのでしょうか？

### DISCUSSION THREAD

**Discussion by 1ft-seabass** [1ft-seabass/ma-hackathon-qa-and-knowledge-202210#8] [Q&A] [answered]
（メンターのサルベージ投稿です）

![2022-11-02 10 10 41](https://user-images.githubusercontent.com/2148840/199371403-79be643d-fcd8-4d70-93aa-8323db87eb43.jpg)

> サーボモータのM5Stack接続側のケーブルは各自で加工するのでしょうか？

という質問がありました。

**Comment by 1ft-seabass** [comment, marked as answer]
Grove ケーブルがつながる PORT A でつなぐやり方があります。ハンズオンではそのように紹介しておりましたー。
https://1ft-seabass.github.io/ma-m5stack-handson-202210/10-servo.html

**Reply by 1ft-seabass** [reply]
加えて、Grove 変換ケーブル（オスピン化するやつ）で M5Stack のサイドにあるメスピン穴に対してつなぐこともできます。

Grove 変換ケーブルは、

GROVE - 4ピン - ジャンパオスケーブル（5本セット） — スイッチサイエンス
https://www.switch-science.com/products/6245

これですね。

たとえば、M5Stack のサイドにあるメスピン穴では

- 赤 → 5V
- 黒 → G
- 黄 → 2 番
- 白 → 挿さない

と挿してみます。

その場合 https://1ft-seabass.github.io/ma-m5stack-handson-202210/10-servo.html のソースコードで 22 のところを、挿したところに合わせて 2 に直す必要がありますので、ご注意ください。

**Reply by miyo2x** [reply]
センサを2つ接続する場合は、PORT.Aがふさがるので、ケーブルの購入が必要ということですよね？
ちなみに超音波距離センサは21に接続とありました。

**Reply by 1ft-seabass** [reply]
@miyo2x はい。この変換ケーブルは必要になります。今回、なんらかのセンサー希望すると、一緒に配布されてませんでしたっけ？違ってたらすいません。

**Reply by 1ft-seabass** [reply]
あ、補足です。サーボはportAを使って、他のセンサーを変換ケーブルをつかって差す感じです！

**Reply by 1ft-seabass** [reply]
@miyo2x ↑

**Reply by miyo2x** [reply]
@1ft-seabass センサにはPORT.Aのケーブルしかないのでダメですね。本体に付属のケーブルは延長ケーブル的なもので変換はできませんでした。考えます...

**Reply by 1ft-seabass** [reply]
おおお、なるほど。そういうことですね！本体付属のケーブルでは厳しいですね。

たしかにハンズオンドキュメントには、センサー系をこの変換ケーブル前提でナレッジ書いていましたが、複数となると色々考える要素がでてきますね。

@miyo2x ちょっと、今日明日で解決できる方法を模索するので、別 Discussions でナレッジお伝えできるようにしますー。

**Reply by 1ft-seabass** [reply]
@miyo2x 参考になるか分かりませんが、まとめましたー。

→ https://github.com/1ft-seabass/ma-hackathon-qa-and-knowledge-202210/discussions/15

**Reply by miyo2x** [reply]
@1ft-seabass ありがとうございました。日曜午前中にに千石電商で購入しておきましたのでハッカソンは無事に進めることができてます。それより複数接続するとイベントのハンドリングが大変でした

**Reply by 1ft-seabass** [reply]
@miyo2x おお！無事進められててよかったです～。

そうですね、複数接続は、どんどん組むの大変になりますよねー。応援してます！
